wordpress-mobile/WordPress-Android
# Issue/9815 email verification reminder

Fixes #9815

This PR email verification Toast after domain registration + some refactoring.

Some work on this PR has been done in #9866

To test:
1. Install the wasabi build.
2. Switch to a site on a premium plan with unclaimed domain credit
3. The CTA should be displayed near the top of the screen, click on "Register Domain"
4. Pick a domain (preferably .blog or .com).
5. Tap on "CHOOSE DOMAIN" and navigate to contact form details.
6. Fill out your contact details (if they haven't already been prefilled)
7. Click "REGISTER DOMAIN"
8. If successful, the Success screen should be displayed. Click on "NEXT". You should now be returned to the **My SIte** screen
9. Verify that the "Verify your email address..." toast message appears when you return to the **My Site** screen

PS: Follow domain testing guideline and make sure if you purchase a domain to cancel it.

![Screenshots2](https://user-images.githubusercontent.com/897223/57747840-7c0d6200-7694-11e9-99e6-1471b2c78699.png)

Update release notes:
